ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client/InFutureCheck.hs

```diff
@@ -136,7 +136,7 @@ realHeaderInFutureCheck skew systemTime =
         unless tooEarly $ do
             now <- systemTimeCurrent systemTime
             let ageNow         = now `diffRelTime` onset
-                syntheticDelay = negate ageNow
+                syntheticDelay = max 0 $ negate ageNow
```

Comment:
Why was `max 0` added? Can `onset` occur after `now`?

Reply:
Thanks for asking! It's trickier than I thought: see https://github.com/input-output-hk/io-sim/issues/129